extension/packagemanagerextension/nbextension/api.js

```diff
@@ -0,0 +1,117 @@
+define([
+    'jquery',
+    './urls'
+], function ($, urls) {
+    "use strict";
+
+    let endpoints = {
+        "update_packages": {
+            "uri": "packages",
+            "method": "PATCH"
+        },
+        "checkupdate": {
+            "uri": "packages/check_update?project=",
+            "method": "GET"
+        },
+        "delete_packages": {
+            "uri": "packages",
+            "method": "DELETE"
+        },
+        "install_packages": {
+            "uri": "packages",
+            "method": "POST"
+        },
+        "getinfo": {
+            "uri": "project_info?project=",
+            "method": "GET"
+        },
+        "search": {
+            "uri": "packages/search?q=",
+            "method": "GET"
+        }
+    };
+
+    /*
+    This function makes the API calls to the specified endpoint with the corresponding headers
+    */
+    function api_call(url, method, payload = "") {
```

Comment:
Why don't you do `api_call(endpoint, url, payload = "")`?
With those you already know the url (`urls.api_url + endpoint.uri + url`) and the method (`endpoint.method`).

Reply:
And payload should be `= {}`. As far as I remember, you don't even need to stringify.. And if you do, just do it here.